### DISCUSSION TITLE
Best way to add a new method

### DISCUSSION THREAD

**Discussion by sreekanthmr** [salman0ansari/whatsapp-api-nodejs#199] [Q&A]
I would like to add a couple of methods such as get read receipt of a specific message (just an example).  What is the best way to achieve that?

Thanks

**Comment by salman0ansari** [comment]
learn [baileys](https://github.com/adiwajshing/Baileys/)
add a function in Instance Class

**Comment by sreekanthmr** [comment]
@salman0ansari thanks for your quick response.

I added a routing, a function in the controller, and finally a function in instance.js.  But I was stuck midway through that.  Of course, it is my ignorance of things are arranged.

I wanted to call fetchMessageReceipts from the instance class but was not able to do that.  I was a bit confused about how the functions in baileys are called from instance class.

Thanks

**Reply by salman0ansari** [reply]
show code

**Reply by sreekanthmr** [reply]
I created a new route
router.route('/readreceipt').get(keyVerify,loginVerify,controller.GetReadReceipt)

Add this in controller
exports.GetReadReceipt = async (req, res) => {
    const data = await WhatsAppInstances[req.query.key]?.getReadReceipt(
        req.query.messageId)
    return res.status(201).json({ error: false, data: data })
}

Wanted to add this to instance but I know it will not work because I don't know how to connect with fetchMessageReceipts function, which is in make-in-memory-store.js

async getReadReceipt(msgKey) {
        this.instance
        const result = await sock.fetchMessageReceipts(msgKey);
        return result
    }

Thanks

**Reply by salman0ansari** [reply]
`fetchMessageReceipts` is not an inbuilt function of baileys.

**Reply by sreekanthmr** [reply]
I do find it in Baileys under lib/store/make-in-memory-store.js

fetchMessageReceipts: async ({ remoteJid, id }, sock) => {
            const list = messages[remoteJid];
            const msg = list === null || list === void 0 ? void 0 : list.get(id);
            let receipts = msg.userReceipt;
            if (!receipts) {
                receipts = await (sock === null || sock === void 0 ? void 0 : sock.messageInfo(remoteJid, id));
                if (msg) {
                    msg.userReceipt = receipts;
                }
            }
            return receipts;
        }

**Reply by salman0ansari** [reply]
you have to fetch `messages array` from db

**Reply by sreekanthmr** [reply]
Is there any existing code I can look at in this repo? just to understand how to do it better.

**Reply by salman0ansari** [reply]
```js
await Chat.findOne({key: this.key})
```

**Reply by sreekanthmr** [reply]
Oh that is really helpful.  I just realized I can use getChat() directly and manipulate the results as per my requirements.

Thanks for your direction

**Reply by sreekanthmr** [reply]
Thanks, I'm able to progress well.  A small request... I'm able to get the chat object but trying to get an individual message.  Pls give some directions to achieve the same.

Thanks
